docs/ocis/adr/0011-global-url-format.md

```diff
@@ -0,0 +1,210 @@
+
+`</namespaced/alias></relative/path/to/resource>` is the global path in the CS3 api. The CS3 Storage Registry is responsible by managing the mount points.
+
+In order to be able to copy and paste URLs all resources must be uniquely identifyable:
+
+* Instead of `/home` the URL always has to reflect the user: `/personal/einstein`
+* Public links can use `/public/<token>`
+* workspaces can use `/workspaces/<alias>` or `/workspaces/<additional>/<classification>/<alias>` where the hierarchy is given by the organization
+* experiments can use `/experiments/<alias>`
+* research institutes could set up `/papers/<researchgroup>/<alias>`
+* trash could be accessed by prefixing the namespace alias with `/trash`? or using `/trash/<space_id>`
+* instead of a namespaced alias a storage space id could be used with a generic `/space/<space_id>` namespace
+
+The alias namespace hierarchy and depth can be pre determined by the admin. Even if aliases change the `id` parameter prevents bookmarks from breaking. A user can decide to build a different hierarchy by using his own registry. 
+
+What about shares? Similar to `/home` it must reflect the user: `/shares/einstein` would list all shares *by* einstein for the currently logged in user. The ui needs to apply the same URL rewriting as for space based URLs: when navigating into a share the URL has to switch from `/personal/einstein/relative/path/to/shared/resource` to `/shares/einstein/<unique and potentially namespaced alias for shared resource>`. When more than one `resource` was shared a name collision would occur. To prevent this we can use ids `/shares/einstein/id/<resource_id` or namespaced aliases `/shares/einstein/files/alias`. Similar to the `/trash` prefix we could treat `/shares` as a filter for the shared resources a user has access to, but that would disclose unshared path segments in personal spaces. We could make that a feature and let users create an alias for a shared resource, similar as for public links. Then they can decide if they want to disclose the full path in their personal space (or another workspace) or if they want to use an alias which is then accessed at `/shares/einstein/<alias>`. As a default we could take the alias at creation time from the filename. That way two shares to a resource with the same name, eg.: `/personal/einstein/project AAA/foo` and `/personal/einstein/project BBB/foo` would lead to `/shares/einstein/foo` (a CS3 internal reference to `/personal/einstein/project AAA/foo`) and `/shares/einstein/foo (2)` (a CS3 internal reference to `/personal/einstein/project BBB/foo`). `foo (2)` would keep its name even when `foo` is deleted or renamed. Well an id as the alias might be better then, because users might rename these aliases, which would break URLs if they have been bookmarked. In any case this would make end user more aware of what they share AND it would allow them to choose an arbitrary context for the links they want to send out: personal internal share URLs. 
```

Comment:
we could also roll an id as the alias but append a description after a `#`...

wait, the aliases for shares still would have an id: `https://demo.owncloud.com/shares/einstein/foo?id=b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21609`, so bookmarks would not break and the alias can change. 😍

Reply:
> but that would disclose unshared path segments in personal spaces

Do we need an example for this? I think is one of these things that look obvious but need some examples

Reply:
I'm having troubles reading this bit, it's a bit dense 😵

Reply:
k, will try to clarify tomorrow